From 3d1a5fa9ab4a38fd5c2004820907158e1e774858 Mon Sep 17 00:00:00 2001
From: Orgad Shaneh <orgad.shaneh@audiocodes.com>
Date: Sat, 4 May 2024 22:49:06 +0300
Subject: [PATCH] RealDiskInterface: Do *not* set locale to an empty string

It causes the cursor handling to be extremely slow on MinGW.

Added in #2321.

Fixes #2435.
---
 src/disk_interface.cc | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/disk_interface.cc b/src/disk_interface.cc
index 5aba2179ef..1259c612f4 100644
--- a/src/disk_interface.cc
+++ b/src/disk_interface.cc
@@ -161,8 +161,6 @@ bool DiskInterface::MakeDirs(const string& path) {
 RealDiskInterface::RealDiskInterface()
 #ifdef _WIN32
 : use_cache_(false), long_paths_enabled_(false) {
-  setlocale(LC_ALL, "");
-
   // Probe ntdll.dll for RtlAreLongPathsEnabled, and call it if it exists.
   HINSTANCE ntdll_lib = ::GetModuleHandleW(L"ntdll");
   if (ntdll_lib) {
